src/wallet/wallet.cpp

```diff
@@ -4453,4 +4470,17 @@ std::optional<CKey> CWallet::GetKey(const CKeyID& keyid) const
     }
     return std::nullopt;
 }
+
+void CWallet::WriteBestBlock() const
+{
+    AssertLockHeld(cs_wallet);
+
+    if (!m_last_block_processed.IsNull()) {
+        CBlockLocator loc;
+        chain().findBlock(m_last_block_processed, FoundBlock().locator(loc));
```

Comment:
In commit "wallet: Update best block record after block dis/connect" (7bacabb204b6c34f9545f0b37e2c66296ad2c0de)

IMO would be good to assert `!loc.IsNull()` as a sanity check to ensure that doesn't happen. (It shouldn't, but node or wallet code could change in the future and invalidate current assumptions.)